src/VirtualClient/VirtualClient.Actions/SysbenchOLTP/SysbenchOLTPExecutor.cs

```diff
@@ -140,24 +140,37 @@ protected void InitializeApiClients()
         /// </summary>
         protected async Task InitializeExecutablesAsync(CancellationToken cancellationToken)
         {
-            if (this.Platform == PlatformID.Unix)
+            // store state with initialization status & record/table counts, if does not exist already
+
+            SysbenchOLTPState state = await this.stateManager.GetStateAsync<SysbenchOLTPState>(nameof(SysbenchOLTPState), cancellationToken)
+                ?? new SysbenchOLTPState();
+
+            if (!state.ExecutablesInitialized)
             {
-                string scriptsDirectory = this.PlatformSpecifics.GetScriptPath("sysbencholtp");
-
-                await this.SystemManager.MakeFileExecutableAsync(
-                    this.Combine(scriptsDirectory, "balanced-server.sh"),
-                    this.Platform,
-                    cancellationToken);
-
-                await this.SystemManager.MakeFileExecutableAsync(
-                    this.Combine(scriptsDirectory, "balanced-client.sh"),
-                    this.Platform,
-                    cancellationToken);
-
-                await this.SystemManager.MakeFileExecutableAsync(
-                    this.Combine(scriptsDirectory, "in-memory.sh"),
-                    this.Platform,
-                    cancellationToken);
+                // install sysbench using repo scripts
+                if (this.Platform == PlatformID.Unix && this.DatabaseScenario != SysbenchOLTPScenario.Default)
+                {
+                    string scriptsDirectory = this.PlatformSpecifics.GetScriptPath("sysbencholtp");
+
+                    await this.SystemManager.MakeFileExecutableAsync(
```

Comment:
There is one single method to make all files under one dir executable if you want